appengine/reconciletags/version_check_test.py

```diff
@@ -33,21 +33,64 @@
                "| egrep -o \"v{}\\.[0-9]+\" | sort | uniq | cut -c 2-"),
     "go1-builder": ("curl -L https://golang.org/dl"
                     "| egrep -o \"go{}\\.[0-9]\" | sort | uniq | cut -c 3-"),
-    "java": ("sudo apt-get -qq install {0}; apt-cache show {0}"
-             "| grep \"Version:\" | awk '{{ print $2 }}'")
+    "java": ("docker run -it --entrypoint /bin/bash {0} "
+             "-c \"apt-get update &> /dev/null; apt-get install -s {1}"
+             "| grep \\\"Conf {2}\\\" | awk '{{ print \\$3 }}' | cut -c 2-\"")
+
+}
+
+runtime_to_current_version = {
+    "aspnetcore": "dotnet --info",
+    "debian": "apt-get update; apt-get -y install lsb-release; lsb_release -a",
+    "ubuntu": "apt-get update; apt-get -y install lsb-release; lsb_release -a",
+    "ruby": "ruby -v",
+    "python": "python3 --version",
+    "php": "php -v",
+    "nodejs": "node --version",
+    "java": "java -version",
+    "go1-builder": "echo $GO_VERSION"
+}
+
+runtime_to_filter = {
```

Comment:
For the parsing that happens outside of the docker command, could you do this in python instead? Instead of filter bash commands, this could be a map of functions?